### PR TITLE
build: update dev deps

### PR DESCRIPTION
**Pull request by branchvincent**
Just some miscellaneous cleanup to:

- Update dev dependencies
- Simplify vendoring config and re-sync
- Address our own [deprecation warning](https://github.com/python-poetry/poetry-core/runs/6442139532?check_suite_focus=true#step:12:22) when running tests

### CODE REVIEW

File: src/poetry/core/_vendor/jsonschema/LICENSE [hunk 1]
```diff
@@ -1,19 +0,0 @@
-Copyright (c) 2012 Julian Berman
```
Comment (branchvincent) [inline src/poetry/core/_vendor/jsonschema/LICENSE:1]: `vendoring` removed this, but note there is already a [`COPYING`](https://github.com/python-poetry/poetry-core/blob/master/src/poetry/core/_vendor/jsonschema/COPYING)

**Comment by branchvincent** [comment]
Downstream test failure is due to https://github.com/pypa/setuptools/issues/3320

**Comment by abravalheri** [comment]
Sorry for the disruption. Soon the new setuptools version will be available with the fix for the unexpected error (it should only be a warning).
Please let me know if it does not work.

File: tests/version/test_version_pep440.py [hunk 2]
```diff
@@ -25,6 +25,7 @@ def test_pep440_release_segment_from_parts(
     assert Release.from_parts(*parts) == result
 
 
+@pytest.mark.filterwarnings("ignore::DeprecationWarning")
```
Comment (abn) [inline tests/version/test_version_pep440.py:28]: Better to remove the `short` parameter.

**Review by abn — CHANGES_REQUESTED** [review]

File: tests/version/test_version_pep440.py (hunk 2 above)
Comment (branchvincent) [reply tests/version/test_version_pep440.py]: oops, of course 😅

**Review by abn — APPROVED** [review]

**Pull request merged**